lib/puma/server.rb

```diff
@@ -424,7 +425,7 @@ def process_client(client, buffer)
 
             ThreadPool.clean_thread_locals if clean_thread_locals
 
-            unless client.reset(@status == :run)
+            unless client.reset(@status == :run && @fast_track_ka_timeout)
```

Comment:
I don't mind pulling it out into a variable because that value will never change for the lifetime of the process. Shouldn't be an instance variable though, just use a local.